raspberry/read_sbus/read_sbus_from_GPIO_receiver.py

```diff
@@ -1,46 +1,92 @@
+parser = RadioDataParser()
+
+# TODO: implement the below loop as a thread with frequency 50Hz. Might want to .join() and manage threads using 
+# ThreadPoolExecutor as a context manager for multi-threading tasks such as altitude hold, loitering algos for future usage.
+# NOTE: When a thread terminates with an exception, the ThreadPoolExecutor will not report it, so might want to handel exceptions properly
+# in the threaded functions
+# NOTE: The order of execution of pooled threads cannot be predicted, this is highly dependent on what the OS does at that time.
+# This will cause many synchronization issues and racing conditions if they use shared resources, 
+# for example let us say that thread_1 needs results of thread_2, so the required order of 
+# execution is start thread_2, let it finish, start thread_1, let it finish. But pooling threads 1 and 2 cannot gurantee which of the two run 
+# first. It beocmes worse if they share resources, as one might not be aware that the other changed some values in the shared resurce (not thread-safe).
+# Possible solution is to use Locks (but make sure to release them). 
+
+
+def get_radio_data():
+        is_connected = reader.is_connected()
+        packet_age = reader.get_latest_packet_age() #milliseconds
+
+        #returns list of length 16, so -1 from channel num to get index
+        channel_data = str(reader.translate_latest_packet())[1:-1]
+
+        return is_connected, packet_age, channel_data
+
+def parse_radio_data(channel_data):
+        # check if data is in range [1000, 2000]
+        parser.m_channelData = list(map(lambda x: 0 if int(x) < 0 else (2000 if int(x) > 2000 else int(x)), channel_data.strip().split(",")))
+        # process and encapsulate the data
+        # the output data packet format will be as follows
+        # Y_radPs, P_rad, R_rad, T_PWM_MIN2MAX, % trimA, % trimB, % trimC, % trimE, encodedSwitchesData
+        # here the final data value encodedSwitchesData is an integer carrying information 
+        # of the position of switches A, B, C, and D in the last 5-bits (the rightmost 5-bits),
+        # in which each bit corresponds to data as follows
+        # bit-4 (MSB): 1-bit info of Switch B: 1 if armed else 0
+        # bit-3: 1-bit info of Switch A: 1 if kill_on else 0
+        # bits 2 and 1: 2-bit info of switch C: 00 for position DOWN, 01 for position MID, 10 for position UP
+        # bit-0: 1-bit info of switch D: 1 if D_on else 0
+        channel_data = parser.encapsulateRadioData()
+        # print(channel_data) #Uncomment to Print data received on Pi from the RC receiver
+        return channel_data
+
+def send_data_to_ESP(channel_data):
+        # Send with S in the beginning to indicate the start of the data, and also useful to check if data 
+        # is received properly on the ESP's end
+        ser.write(f'S,{channel_data}\n'.encode()) #Send data from Pi to ESP32
+        # ser.write(b"\n") #Starts a new line so ESP32 knows when to stop reading
```

Comment:
Remove all commented-out code

Reply:
Done. Left out the ones that have prompts like `#uncomment to do something`.